### PR TITLE
Add r-sodium

### PR DESCRIPTION
**Pull request by halldc**
Adding a recipe for the [sodium](https://cran.r-project.org/package=sodium) R package. This was created using the [conda_r_skeleton_helper](https://github.com/bgruening/conda_r_skeleton_helper), and then I added the `libsodium` requirement.

### CODE REVIEW

File: recipes/r-sodium/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    - {{native}}toolchain        # [win]
+    - {{posix}}filesystem        # [win]
+    - {{posix}}sed               # [win]
+    - {{posix}}grep              # [win]
+    - {{posix}}autoconf
+    - {{posix}}automake          # [not win]
+    - {{posix}}automake-wrapper  # [win]
+    - {{posix}}pkg-config
+    - {{posix}}make
+    - {{posix}}sed  # [win]
```
Comment (cbrueffer) [inline recipes/r-sodium/meta.yaml]: sed and coreutils are duplicated

Comment (halldc) [reply recipes/r-sodium/meta.yaml]: Good catch! Looks like that's a bug in the conda_r_skeleton_helper.
Anyway, looks like it's passed tests again.

File: recipes/r-sodium/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    - {{posix}}autoconf
+    - {{posix}}automake          # [not win]
+    - {{posix}}automake-wrapper  # [win]
+    - {{posix}}pkg-config
+    - {{posix}}make
+    - {{posix}}coreutils         # [win]
+    - {{posix}}zip               # [win]
+  host:
+    - r-base
+    - libsodium >=1.0.3
```
Comment (cbrueffer) [inline recipes/r-sodium/meta.yaml]: Shouldn't libsodium be a run-depend as well?

Comment (halldc) [reply recipes/r-sodium/meta.yaml]: It uses `run_exports` (see [here](https://github.com/conda-forge/libsodium-feedstock/blob/master/recipe/meta.yaml#L14)), so we don't need to add it to the run requirements.

**Pull request merged**